File: Python3/program101.py
# IntPrompt() function
def IntPrompt( text ):
    goodParse = False
    while not goodParse:
        try:
            result = int(input(text))
            goodParse = True
        except:
            print('*Invalid input. Try again*')
    return result

# Main script
count = IntPrompt('How many grades will you enter: ')
while count < 3:
    print('*Minimum value of 3 expected*')
    count = IntPrompt('How many grades will you enter: ')

print()
grades = []

# input grades
for i in range(count):
    grades.append(IntPrompt('Enter grade %d: ' %(i+1)))

# print grades
print('\nYour grades:')
found = False
for g in grades:
    if g == min(grades) and not found:
        print('%d *dropped*' %(g))
        found = True
    else:
        print('%d' %(g))

# print average
print('\nAverage of grades is %.2f' %((sum(grades)-min(grades))/(len(grades)-1)))
        

File: Csharp/Program101.cs
using System;

namespace Program101
{
    class Program
    {
        static int IntPrompt(string text)
        {
            bool goodParse = false;
            int result = 0;
            while (!goodParse)
            {
                try
                {
                    Console.Write(text);
                    result = Convert.ToInt32(Console.ReadLine());
                    goodParse = true;
                }
                catch
                {
                    Console.WriteLine("*Invalid input. Try again*");
                }
            }

            return result;
        }

        static void Main(string[] args)
        {
            // Prompt for grades
            int count = IntPrompt("How many grades will you enter: ");

            while (count < 3)
            {
                Console.WriteLine("*Minimum value of 3 expected*");
                count = IntPrompt("How many grades will you enter: ");
            }

            int[] grades = new int[count];
            Console.WriteLine();

            int minGrade = int.MaxValue;
            int minIndex = 0;
            int sumGrade = 0;

            for (int i = 0; i < grades.Length; i++)
            {
                grades[i] = IntPrompt($"Enter grade {i + 1}: ");
                sumGrade += grades[i];
                if (grades[i] < minGrade)
                {
                    minGrade = grades[i];
                    minIndex = i;
                }
            }

            // Print entered grades:
            Console.WriteLine("\nYour grades:");

            for (int i = 0; i < grades.Length; i++)
            {
                if (i == minIndex)
                {
                    Console.WriteLine("{0} *dropped*", grades[i]);
                }
                else
                {
                    Console.WriteLine("{0}", grades[i]);
                }
            }

            Console.WriteLine("\nAverage of grades is {0:F}", (sumGrade - minGrade) / (double)(grades.Length - 1));

            Console.Write("\nPress any key to exit...");
            Console.ReadKey();
        }
    }
}
